Mathlib/Data/Option/Basic.lean

```diff
@@ -3,28 +3,370 @@ Copyright (c) 2017 Mario Carneiro. All rights reserved.
 Released under Apache 2.0 license as described in the file LICENSE.
 Authors: Mario Carneiro
 -/
-import Mathlib.Init.Function
-import Mathlib.Data.Option.Defs
-import Mathlib.Logic.Basic
+import Mathlib.Logic.IsEmpty
+import Mathlib.Tactic.Basic
+import Mathlib.Logic.Relator
+import Mathlib.Tactic.SimpTrace
+import Mathlib.Mathport.Rename
```

Comment:
```suggestion
import Mathlib.Logic.IsEmpty
import Mathlib.Logic.Relator
import Mathlib.Tactic.Basic
import Mathlib.Tactic.SimpTrace
import Mathlib.Mathport.Rename
```
Generally we sort imports so tactics come at the end, I think.

We really should put `Mathport.Rename`, `Tactic.SimpTrace`, `Util.WhatsNew`, `Tactic.LibrarySearch`, and possibly some others into a kitchen-sink import, and then import them as low as possible, e.g. `Logic.Basic`, so others don't have to deal with them.